### DISCUSSION TITLE
Trying to update select field options, when another field is updated.

### DISCUSSION THREAD

**Discussion by DanjBethel** [filamentphp/filament#5877] [Help]
Here is my code below. locations is a variable on my livewire component that contains the form schema

```
 Toggle::make('floridaOnly')
                        ->default(true)
                        ->reactive()
                        ->afterStateUpdated(function (Closure $set, $state) {
                            $this->locations = AlamoLocationsFormatter::getFilterOptions(
                                SearchAlamoVehicleLocations::handle()->body(), $state
                            );
                        })
                        ->columnSpan(3),
                    Select::make('fromLocation')
                        ->options($this->locations)
                        ->searchable()
                        ->required(),
```

Not sure if "after state updated" is where this type of logic should go. But basically, I send the state of the "floridaOnly" field in (true or false) and based on that i run a filter and return the locations. The locations value is being changed by this afterStateUpdated hook, but the options in the below select field "fromLocation" aren't being updated.

**Comment by danharrin** [comment]
You dont need to store the options, you just pass a closure to the `options()` method. https://filamentphp.com/docs/2.x/forms/fields#dependant-selects

**Reply by DanjBethel** [reply]
Hey thanks for the reply, the issue im having is that the options are to be grabbed from an external source, however, I have two Select inputs that use the exact same options. This is why I wanted to store them, to circumvent having to make two API calls to get the same data. Any suggestions? Thanks again for your time.

**Reply by danharrin** [reply]
Try `->options(fn () => $this->locations)` instead

**Reply by DanjBethel** [reply]
Works perfectly. This is close to what I had before but I guess the callback to "options" makes all the difference. Thanks again for all that you do. Can't thank you enough!
